airflow/contrib/hooks/aws_hook.py

```diff
@@ -151,3 +151,19 @@ def get_resource_type(self, resource_type, region_name=None):
         session, endpoint_url = self._get_credentials(region_name)
 
         return session.resource(resource_type, endpoint_url=endpoint_url)
+
+    def get_session(self, region_name=None):
+        """Get the underlying boto3.session."""
+        session, _ = self._get_credentials(region_name)
+        return session
+
+    def get_credentials(self, region_name=None):
```

Comment:
If the use-case here is primarily using the credentials to create a session, I would opt to not have a public method that exposes a private method

Reply:
I don't think this exposes the private method. While the private method returns the `session` and `endpoint_url` here, the public one simplifies getting the `credentials` object from the `boto3.session`.
These credentials are useful for copying to a Redshift database:
```python
aws_hook = AwsHook()
pg_hook = PostgresHook('redshift_default')
query = """COPY table FROM 's3://some_bucket/some_key' CREDENTIALS
'aws_access_key_id={c.acces_key};aws_secret_access_key={c.secret_key};token={c.token}'
""".format(c=aws_hook.get_credentials())
pg_hook.run(query)
```

Or when using [pyAthena](https://pypi.python.org/pypi/PyAthena/):
```python
from pyathena import connect
aws_hook = AwsHook()
credentials = aws_hook.get_credentials()
connection = connect(aws_access_key_id=credentials.acces_key,
    aws_secret_access_key=credentials.secret_key,
    s3_staging_dir='s3://YOUR_S3_BUCKET/path/to/')
...
```

In both instances we could use the `session` to get the credentials, but this helps simplify this for the user.

The `get_session()` can be useful if someone wants to create a [boto3 client](http://boto3.readthedocs.io/en/latest/reference/core/session.html#boto3.session.Session.client) with options that the `get_client()` don't allow, e.g. a `config`.

Reply:
Ah gotcha, that makes sense! I wonder if it makes sense at some point to extend the Postgres Hook into a RedshiftDBHook - there are a number of Redshift specific commands that would benefit from abstracting common patterns.

Reply:
Yeah, that sounds like a great idea. Totally something I can find time to pick up!